src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -104,7 +104,7 @@ private function completeDefinition($id, Definition $definition)
                 // Typehint against a non-existing class
 
                 if (!$parameter->isDefaultValueAvailable()) {
-                    continue;
+                    throw new RuntimeException(sprintf('Cannot auto-wire argument %s for %s because the type-hinted class does not exist.', $index + 1, $definition->getClass()));
```

Comment:
The error is better as it will say "Cannot autowire argument 2..." if it really is the *second* argument. That's consistent with PHP errors (e.g. "Missing argument 2 for ...")